force-app/main/default/classes/SOQL.cls

```diff
@@ -108,7 +133,8 @@ public with sharing class SOQL {
      * @return A list of Opportunity records.
      */
     public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
-        return null; // Replace null with the variable you used to store the result
+        List<Opportunity> opp = [SELECT Id, Name, StageName, CloseDate FROM Opportunity WHERE Name LIKE '%keyword'];
```

Comment:
This where your test is failing. You are using the variable here, instead using the string 'keyword'